ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateRoute/NavigateRoute.cpp

```diff
@@ -0,0 +1,271 @@
+  connectRouteTaskSignals();
+
+  m_routeTask->load();
+
+  // add graphics for the predefined stops
+  SimpleMarkerSymbol* stopSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Diamond, Qt::red, 20, this);
+  m_routeOverlay->graphics()->append(new Graphic(conventionCenterPoint, stopSymbol, this));
+  m_routeOverlay->graphics()->append(new Graphic(memorialPoint, stopSymbol, this));
+  m_routeOverlay->graphics()->append(new Graphic(aerospaceMuseumPoint, stopSymbol, this));
+
+  emit mapViewChanged();
+}
+
+void NavigateRoute::connectRouteTaskSignals()
+{
+  connect(m_routeTask, &RouteTask::solveRouteCompleted, this, [this](QUuid, RouteResult routeResult)
+  {
+    if (routeResult.isEmpty())
+      return;
+    m_routeResult = routeResult;
+    m_route = m_routeResult.routes()[0];
+
+    m_mapView->setViewpointGeometry(m_route.routeGeometry(), 100);
+
+    // create a graphic to show the route
+    m_routeAheadGraphic = new Graphic(m_route.routeGeometry(), new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, Qt::blue, 5, this), this);
+
+    // create a graphic to represent the route that's been traveled (initially empty).
+    m_routeTraveledGraphic = new Graphic(this);
+    m_routeTraveledGraphic->setSymbol(new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::cyan, 3, this));
+    m_routeOverlay->graphics()->append(m_routeAheadGraphic);
+    m_routeOverlay->graphics()->append(m_routeTraveledGraphic);
+
+    m_navigationButtonEnabled = true;
+    emit navigationButtonChanged();
+  });
+
+  connect(m_routeTask, &RouteTask::createDefaultParametersCompleted, this, [this](QUuid, RouteParameters defaultParameters)
+  {
+    // set values for parameters
+    defaultParameters.setReturnStops(true);
+    defaultParameters.setReturnDirections(true);
+    defaultParameters.setReturnRoutes(true);
+    defaultParameters.setOutputSpatialReference(SpatialReference::wgs84());
+
+    Stop stop1(conventionCenterPoint);
+    Stop stop2(memorialPoint);
+    Stop stop3(aerospaceMuseumPoint);
+
+    QList<Stop> stopsList = {stop1, stop2, stop3};
+    defaultParameters.setStops(stopsList);
+
+    m_routeTask->solveRoute(defaultParameters);
+  });
+
+  connect(m_routeTask, &RouteTask::doneLoading, this, [this](Error error)
+  {
+    if (error.isEmpty())
+    {
+      m_routeTask->createDefaultParameters();
+    }
+    else
+    {
+     qDebug() << error.message() << error.additionalMessage();
+    }
+  });
+}
+
+bool NavigateRoute::navigationButtonEnabled() const
+{
+  return m_navigationButtonEnabled;
+}
+
+bool NavigateRoute::recenterButtonEnabled() const
+{
+  return m_recenterButtonEnabled;
+}
+
+void NavigateRoute::startNavigation()
+{
+  // disable the navigation button
+  m_navigationButtonEnabled = false;
+  emit navigationButtonChanged();
+
+  // get the directions for the route
+  m_directions = m_route.directionManeuvers(this);
+
+  // create a route tracker
+  m_routeTracker = new RouteTracker(m_routeResult, 0, this);
```

Comment:
if this can happen more than once (e.g. every time the user click's a button) then this will leak

Reply:
The button can only be clicked once